packages/devtools/lib/src/inspector/inspector_controller.dart

```
@@ -734,4 +757,27 @@ class InspectorController implements InspectorServiceClient {
     }
     parent?.maybeUpdateValueUI(valueRef);
   }
+
+  Future<void> expandAllNodesInDetailsTree() async {
+    await details.recomputeTreeRoot(
+      inspectorTree.selection.diagnostic,
+      details.inspectorTree.selection.diagnostic,
+      false,
+      subtreeDepth: maxJsInt,
+    );
+  }
+
+  Future<void> resetDetailsTree() async {
+    assert(details.inspectorTree.selection?.diagnostic != null);
```

there might not be a selection. In that case we should fail fast.